fendermint/app/src/main.rs

```diff
@@ -30,6 +31,91 @@ async fn main() {
     }
 }
 
+fn create_log(level: tracing::Level, log_dir: Option<&PathBuf>) -> anyhow::Result<()> {
+    let console_appender = config::Appender::Console;
+
+    // default logging output info log to console
+    let default = config::Logger {
+        level: config::LevelFilter::from_str(level.as_str())?,
+        appenders: literally::hset! { AppenderId::from("console") },
+        format: Format::default(),
+    };
+
+    let mut config = Config {
+        default,
+        loggers: literally::hmap! {},
+        appenders: literally::hmap! {AppenderId::from("console") => console_appender},
+    };
+
+    if let Some(log_dir) = log_dir {
+        let log_folder = expand_tilde(log_dir)
+            .to_str()
+            .ok_or_else(|| anyhow!("cannot parse log folder"))?
+            .to_string();
+        std::fs::create_dir_all(&log_folder).context("cannot create log folder")?;
+
+        let topdown_appender = config::Appender::RollingFile {
+            path: format!("{log_folder}/topdown.log"),
+            policy: Policy {
+                maximum_file_size: "10mb".to_string(),
+                // we keep the last 5 log files, older files will be deleted
+                max_size_roll_backups: 5,
+                pattern: None,
+            },
+        };
+        let debug_appender = config::Appender::RollingFile {
+            path: format!("{log_folder}/debug.log"),
+            policy: Policy {
+                maximum_file_size: "10mb".to_string(),
+                // we keep the last 5 log files, older files will be deleted
+                max_size_roll_backups: 5,
+                pattern: None,
+            },
+        };
+
+        let topdown_logger = config::Logger {
+            level: config::LevelFilter::DEBUG,
+            appenders: literally::hset! { AppenderId::from("topdown") },
+            format: Format::default(),
+        };
+        let debug_logger = config::Logger {
+            level: config::LevelFilter::DEBUG,
+            appenders: literally::hset! { AppenderId::from("debug") },
+            format: Format::default(),
+        };
+
+        config.loggers.insert(
+            Target::from("fendermint_vm_topdown"),
+            topdown_logger.clone(),
+        );
+        config.loggers.insert(
+            Target::from("fendermint_vm_interpreter::chain"),
+            topdown_logger,
+        );
+        config
+            .appenders
+            .insert(AppenderId::from("topdown"), topdown_appender);
+
+        // maybe there are better ways to handle this, but * does not seem to be supported
+        config
+            .loggers
+            .insert(Target::from("fendermint_app"), debug_logger.clone());
+        config
+            .loggers
+            .insert(Target::from("fendermint_vm_interpreter"), debug_logger);
```

Comment:
https://docs.rs/trace4rs/0.5.1/src/trace4rs/handle/loggers.rs.html#91 suggests that a simple `Target::from("fendermint")` should work.

Reply:
funny thing is, with this, all logs are directly to `debug` and `console` is empty

Reply:
Maybe because it's set as a "default" which could mean "if nothing else matched"? Perhaps insert it explicitly with the "" target?

Reply:
Yes :point_right:  https://docs.rs/trace4rs/0.5.1/src/trace4rs/handle/layers.rs.html#139-149